### PR TITLE
BAEL-6566: Manage Kafka Consumer Groups

### PR DESCRIPTION
**Pull request by pedrolopes9-7**

### CODE REVIEW

File: spring-kafka-2/src/main/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/KafkaConsumerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.baeldung.spring.kafka.managingkafkaconsumergroups;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.DoubleDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafka;
```
Comment (JavierPeris) [inline spring-kafka-2/src/main/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/KafkaConsumerConfiguration.java:9]: You can remove this import.

File: spring-kafka-2/src/test/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/ManagingConsumerGroupsIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Autowired
+    MessageConsumerService consumerService;
+
+    @Test
+    public void givenContinuousMessageFlow_whenAConsumerLeavesTheGroup_thenKafkaTriggersPartitionRebalance() throws InterruptedException {
+        int currentMessage = 0;
+        final String consumer1Identifier = "org.springframework.kafka.KafkaListenerEndpointContainer#1";
+        final int totalProducedMessages = 50000;
+        final int messageWhereConsumer1LeavesGroup = 10000;
```
Comment (JavierPeris) [inline spring-kafka-2/src/test/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/ManagingConsumerGroupsIntegrationTest.java:34]: If there's no very specific reason we avoid the use of final in code (we all know that immutability objects have several benefits, but for demonstration code, we favor simplicity/brevity over immutability).

In fact, I'm thinking we can move those values out of the method and declare them as constant, what do you think? In that case, we reduce even more the code for the article.

Comment (pedrolopes9-7) [reply spring-kafka-2/src/test/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/ManagingConsumerGroupsIntegrationTest.java:34]: I liked the idea of global constants

File: spring-kafka-2/src/test/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/ManagingConsumerGroupsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest(classes = ManagingConsumerGroupsApplicationKafkaApp.class)
+@EmbeddedKafka(partitions = 2, brokerProperties = {"listeners=PLAINTEXT://localhost:9092", "port=9092"})
+public class ManagingConsumerGroupsIntegrationTest {
+
+    private static final String consumer1Identifier = "org.springframework.kafka.KafkaListenerEndpointContainer#1";
+    private static final int totalProducedMessages = 50000;
+    private static final int messageWhereConsumer1LeavesGroup = 10000;
```
Comment (JavierPeris) [inline spring-kafka-2/src/test/java/com/baeldung/spring/kafka/managingkafkaconsumergroups/ManagingConsumerGroupsIntegrationTest.java:22]: Just a style issue: The naming convention for constants is uppercase with words separated by underscores.

**Pull request merged**